sdk/app_cpu1/common/drv/eddy_current_sensor.c

```diff
@@ -9,36 +9,101 @@ void eddy_current_sensor_init(void)
 {
     printf("EDDY CURRENT SENSOR:\tInitializing...\n");
 
-    // Set sampling rate to 20kHz
-    eddy_current_sensor_set_sample_rate(EDDY_CURRENT_SENSOR_1_BASE_ADDR, 20000);
+    // Default PWM carrier frequency for AMDC is 100 kHz
+    //   Triggering on both PWM high and low would be an effective sample frequency of 200 kHz,
+    //   either high OR low, being effectively 100 kHz sampling.
+    //   To meet this timing, SCLK would need to be VERY fast.
+    //   Therefore, set eddy current sensors to only sample on PWM high by default
+    //   with an SCLK frequency of only 2 MHz. Because this won't fit into the 10us interval,
+    //   every other PWM trigger will be ignored for an effective sampling rate of 50 kHz
+    eddy_current_sensor_trigger_on_pwm_high(EDDY_CURRENT_SENSOR_1_BASE_ADDR);
+    eddy_current_sensor_set_timing(EDDY_CURRENT_SENSOR_1_BASE_ADDR, 2000, 270);
 
 #if USER_CONFIG_HARDWARE_TARGET == AMDC_REV_E
-    eddy_current_sensor_set_sample_rate(EDDY_CURRENT_SENSOR_2_BASE_ADDR, 20000);
-    eddy_current_sensor_set_sample_rate(EDDY_CURRENT_SENSOR_3_BASE_ADDR, 20000);
-    eddy_current_sensor_set_sample_rate(EDDY_CURRENT_SENSOR_4_BASE_ADDR, 20000);
+    eddy_current_sensor_trigger_on_pwm_high(EDDY_CURRENT_SENSOR_2_BASE_ADDR);
+    eddy_current_sensor_set_timing(EDDY_CURRENT_SENSOR_2_BASE_ADDR, 2000, 270);
+    eddy_current_sensor_trigger_on_pwm_high(EDDY_CURRENT_SENSOR_3_BASE_ADDR);
+    eddy_current_sensor_set_timing(EDDY_CURRENT_SENSOR_3_BASE_ADDR, 2000, 270);
+    eddy_current_sensor_trigger_on_pwm_high(EDDY_CURRENT_SENSOR_4_BASE_ADDR);
+    eddy_current_sensor_set_timing(EDDY_CURRENT_SENSOR_4_BASE_ADDR, 2000, 270);
```

Comment:
Any simple way to put this in a for loop for better readability?

e.g.,

```C
uint32_t base_addrs[3] = {
  EDDY_CURRENT_SENSOR_2_BASE_ADDR,
  EDDY_CURRENT_SENSOR_3_BASE_ADDR,
  EDDY_CURRENT_SENSOR_4_BASE_ADDR,
};

for (int i = 0; i < 3; i++) {
    eddy_current_sensor_trigger_on_pwm_high(base_addrs[i]);
    eddy_current_sensor_set_timing(base_addrs[i], 2000, 270);
}
```

...eh this is probably more lines of code. if you like this, feel free to adopt. Or, maybe you can just reorder the function calls you have so the same calls are next to each other? that would help readability